tests/electronic_structure/test_plotter.py

```diff
@@ -317,7 +317,7 @@ def test_plot_complexity_factor_mu(self):
 
     def test_plot_conductivity_dop(self):
         ax = self.plotter.plot_conductivity_dop()
-        assert len(ax.get_lines()) == 8, "wrong number of lines"
+        assert len(ax.get_lines()) == 16, f"wrong number of lines, got {len(ax.get_lines())}"
```

Comment:
@janosh Perhaps you could help me verify eb7978d3db984786b05fd2e65882a1a992a83a80 and f6a3fc75c425269bdf6a2c9f1af9a79631f6d674? #3237 seems to have broken some tests, and these commits just "fix" those tests, but I'm not sure if it's fixed properly.

Reply:
Sorry I missed some, a8d04f93287c9673495aef494b6284430a4a6efe and 1d104cb22eadc4526f0381aea4a85193613a9df2 might need a closer look as well.

Reply:
there's definitely something broken here. just tried running some of the plot functions. looks like the lines aren't applied to the right axes

![plot_seebeck_dop](https://github.com/user-attachments/assets/771d907f-9ff0-446b-94ec-ed29f8d68c1f)

Reply:
thanks for checking, i suspected the same! the only good news is the fix shouldn't require much code changes...